src/js/elements/menu.js

```diff
@@ -1,174 +1,194 @@
-import {SubIndicator} from '../dataobjects'
-import {checkIfSubCategoryHasChildren, checkIfCategoryHasChildren, Component} from '../utils'
+import { Component, Observable } from '../utils';
 
 const hideondeployClsName = 'hideondeploy';
 const parentContainer = $(".data-mapper-content__list");
 const categoryTemplate = $(".data-category")[0].cloneNode(true);
 const subCategoryTemplate = $(".data-category__h2", categoryTemplate)[0].cloneNode(true);
-const indicatorTemplate = $(".data-category__h2_content", subCategoryTemplate)[0].cloneNode(true);
 const indicatorItemTemplate = $(".data-category__h4", subCategoryTemplate)[0].cloneNode(true);
 const noDataWrapperClsName = 'data-mapper-content__no-data';
 const loadingClsName = 'data-mapper-content__loading';
 
-function subindicatorsInCategory(category) {
-    let count = 0;
-    let subcategories = Object.values(category.subcategories);
-    for (const idx in subcategories) {
-        let subcategory = subcategories[idx];
-        count += subindicatorsInSubCategory(subcategory);
-    }
 
-    return count;
-}
+class LoadMenuModel extends Observable {
+  static EVENTS = {}
 
-function subindicatorsInSubCategory(subcategory) {
+  _subindicatorsInIndicator(groups) {
+    return groups.length;
+  }
 
-    let count = 0;
+  subindicatorsInSubCategory(subcategory) {
     const indicators = Object.values(subcategory.indicators);
+    let count = 0;
     if (indicators.length > 0) {
-        for (const idx in indicators) {
-            let indicator = indicators[idx];
-            count += subindicatorsInIndicator(indicator);
-        }
+      for (const idx in indicators) {
+        let groups = indicators[idx].metadata.groups;
+        count += this._subindicatorsInIndicator(groups);
+      }
     }
 
     return count;
-}
+  }
+
+  checkIfSubCategoryHasChildren(detail) {
+    let hasChildren = false;
+    for (const [_, data] of Object.entries(detail.indicators)) {
+      if (!hasChildren && typeof data.child_data !== 'undefined') {
+        for (const [_, arr] of Object.entries(data.child_data)) {
```

Comment:
same as above - you might want to consider `.values`

Reply:
Done.